hop/io.py

```diff
@@ -111,11 +114,32 @@ def open(self, url, mode="r", metadata=False):
             raise ValueError("mode must be either 'w' or 'r'")
 
 
-class Deserializer(Enum):
-    CIRCULAR = models.GCNCircular
-    VOEVENT = models.VOEvent
-    BLOB = models.MessageBlob
+def _load_deserializer_plugins():
+    """Load all registered deserializer plugins.
+
+    """
+    # set up plugin manager
+    manager = pluggy.PluginManager("hop")
+    manager.add_hookspecs(plugins)
+
+    # load in base models
+    manager.register(models)
+
+    # load external models
+    try:
+        manager.load_setuptools_entrypoints("hop_plugin")
+    except Exception:
+        logger.warning("WARNING: could not load external hop models, verify installed plugins")
```

Comment:
Please remove the `WARNING: ` prefix. The logger should say WARN for you anyway.

I think this message will be pretty cryptic for most users. What does "verify installed plugins" mean? Picture a stumped user getting that message; I don't think they'd know what to do.

Reply:
Those are good points, I'll think about a better way of wording that so that a user will know what's going on.